### PR TITLE
fix #2343 migrate reactor-tests' tests to JUnit 5

### LINKED ISSUE #2343: Migrate reactor-test to use JUnit 5 for its tests
Reactor is currently using both JUnit 4 and JUnit 5 for its own tests. This is a task to harmonize everything to the latest JUnit 5.

This has no impact on user facing code, this is just about tests the project itself uses.

This issue is for reactor-test only because migration is quite a big task. This is the first step, reactor-core comes later

### PR DESCRIPTION
**Pull request by ericbottard**
Getting 
```
Gradle Test Run :reactor-test:test Results: SUCCESS (358 tests, 357 successes, 0 failures, 1 skipped)
```

both before and after this refactoring.

Forward porting to 3.3.x and 3.4.x will require additional work for tests that only exist there.

### CODE REVIEW

File: reactor-test/src/test/java/reactor/test/StepVerifierTests.java [hunk 1]
```diff
@@ -483,7 +484,7 @@ public void thenCancel_cancelsAfterFirst() {
 		                            .then(() -> Schedulers.elastic().schedule(() ->
 				                            publisher.next(1L)))
 		                            .then(() -> Schedulers.elastic().schedule(() ->
-				                            publisher.next(2L), 50, TimeUnit.MILLISECONDS))
+				                            publisher.next(2L), 50, MILLISECONDS))
```
Comment (bsideup) [inline reactor-test/src/test/java/reactor/test/StepVerifierTests.java:487]: let's not import enum values please

File: reactor-test/src/test/java/reactor/test/scheduler/VirtualTimeSchedulerTests.java [hunk 2]
```diff
@@ -23,16 +23,21 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
 import reactor.core.publisher.Mono;
 import reactor.core.scheduler.Scheduler;
-import reactor.core.scheduler.Schedulers;
 import reactor.test.StepVerifier;
 import reactor.test.util.RaceTestUtils;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static reactor.core.scheduler.Schedulers.newElastic;
```
Comment (bsideup) [inline reactor-test/src/test/java/reactor/test/scheduler/VirtualTimeSchedulerTests.java:34]: could you please revert this change? IMO we should only import test util methods.

**Comment by ericbottard** [comment]
Rebased on updated 3.2.x and addressed review comments.

File: reactor-test/src/test/java/reactor/test/StepVerifierTests.java [hunk 3]
```diff
@@ -1114,7 +1116,8 @@ public void verifyVirtualTimeOnErrorAsync() {
 		            .verify();
 	}
 
-	@Test(timeout = 1000)
+	@Test
+@Timeout(value = 1000, unit = TimeUnit.MILLISECONDS)
```
Comment (bsideup) [inline reactor-test/src/test/java/reactor/test/StepVerifierTests.java:1120]: the indentation seems to be wrong:
Screenshot 2020-09-07 at 5 04 57 PM

mixed tabs/spaces?

**Review by bsideup — APPROVED** [review]
approved with a minor nit re the indentation

File: reactor-test/src/test/java/reactor/test/StepVerifierTests.java (hunk 3 above)
Comment (ericbottard) [reply reactor-test/src/test/java/reactor/test/StepVerifierTests.java:1120]: Will fix and propagate same kind of changes to 3.3.x and master (which may have more tests).

Thanks for the review!
